hypothesis-python/src/hypothesis/core.py

```diff
@@ -1007,35 +1015,14 @@ def _execute_once_for_engine(self, data: ConjectureData) -> None:
         """
         trace: Trace = set()
         try:
-            # this is actually covered by our tests, but only on >= 3.12.
-            if (
-                sys.version_info[:2] >= (3, 12)
-                and sys.monitoring.get_tool(MONITORING_TOOL_ID) is not None
-            ):  # pragma: no cover
-                warnings.warn(
-                    "avoiding tracing test function because tool id "
-                    f"{MONITORING_TOOL_ID} is already taken by tool "
-                    f"{sys.monitoring.get_tool(MONITORING_TOOL_ID)}.",
-                    HypothesisWarning,
-                    # I'm not sure computing a correct stacklevel is reasonable
-                    # given the number of entry points here.
-                    stacklevel=1,
-                )
-
```

Comment:
This is moved out of the test execution, to avoid unnecessary difference in behaviour between `execute_once` and `_execute_once_for_engine`